### PR TITLE
Add cryptodocs for the DCGM snap

### PR DESCRIPTION
**Pull request by Deezzir**

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -63,6 +63,13 @@ description: |
 
   The DCGM snap does not currently include the performance testing tool `dcgmproftester` in order to limit the size of the snap.
 
+  **Cryptography**
+  ---
+
+  The snap installs the CUDA keyring, verifying its integrity using `curl` and SHA256 checksums.
```
Comment (aieri) [inline snap/snapcraft.yaml:69]: I think you should be a bit more specific here: we don't verify the integrity of the CUDA keyring, but rather the integrity of the CUDA keyring deb package (which contains the keyring)

Comment (aieri) [inline snap/snapcraft.yaml:69]: how is curl verifying the integrity of the download?

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -63,6 +63,13 @@ description: |
 
   The DCGM snap does not currently include the performance testing tool `dcgmproftester` in order to limit the size of the snap.
 
+  **Cryptography**
```
Comment (aieri) [inline snap/snapcraft.yaml:66]: I think it's worth mentioning how the CUDA keyring is used (i.e. to verify the signature of the DCGM deb package)

File: snap/snapcraft.yaml (hunk 1 above)
Comment (aieri) [inline snap/snapcraft.yaml:69]: careful that this makes it sound like the snap is going to deploy the keyring and the new source on the host system

**Review by aieri — CHANGES_REQUESTED** [review]

File: snap/snapcraft.yaml [hunk 3]
```diff
@@ -66,8 +66,8 @@ description: |
   **Cryptography**
   ---
 
-  The snap installs the CUDA keyring, verifying its integrity using `curl` and SHA256 checksums.
-  The keyring also sets up the appropriate source for the DCGM deb package.\
+  The snap installs the CUDA keyring deb package using `curl` and verifies its integrity using SHA256 checksums.
```
Comment (aieri) [inline snap/snapcraft.yaml:69]: this isn't really accurate: the snap does not install the keyring, in fact as a user installing the DCGM snap I don't get the keyring at all

Comment (aieri) [inline snap/snapcraft.yaml:69]: nit: we install via dpkg, not curl

File: snap/snapcraft.yaml [hunk 4]
```diff
@@ -66,8 +66,8 @@ description: |
...
+  During the snap build process, the keyring sets up the appropriate source for the DCGM deb package and verifies its signature.\
```
Comment (aieri) [inline snap/snapcraft.yaml:70]:
```suggestion
  During the snap build process, the install scripts of the CUDA keyring deb package set up the appropriate source for the DCGM deb package, whose signature is then verified by using the keyring itself.\
```
...although I would suggest rephrasing the whole paragraph so it's clearer that all of this crypto only happens during building

File: snap/snapcraft.yaml (hunk 3 above)
Comment (aieri) [inline snap/snapcraft.yaml:69]: you could also mention that the download happens over https

**Review by aieri — CHANGES_REQUESTED** [review]

File: snap/snapcraft.yaml (hunk 3 above)
Comment (Deezzir) [reply snap/snapcraft.yaml:69]: But the keyring is installed using `curl`

Comment (gabrielcocenza) [reply snap/snapcraft.yaml]: We fetch a debian pkg named `cuda-keyring_1.1-1_all.deb`, check the SHA256SUM and after installing the deb package that the sources are going to change

**Review by gabrielcocenza — COMMENTED** [review]
I think it worth mention that the DCGM exporter can have [TLS or basic-auth](https://github.com/NVIDIA/dcgm-exporter?tab=readme-ov-file#tls-and-basic-auth), but we are not doing this

**Comment by Deezzir** [comment]
> I think it worth mention that the DCGM exporter can have [TLS or basic-auth](https://github.com/NVIDIA/dcgm-exporter?tab=readme-ov-file#tls-and-basic-auth), but we are not doing this

Do we really need to add this, as we don't support that option?

File: snap/snapcraft.yaml [hunk 5]
```diff
@@ -63,6 +63,13 @@ description: |
 
   The DCGM snap does not currently include the performance testing tool `dcgmproftester` in order to limit the size of the snap.
 
+  **Cryptography**
+  ---
+
+  During the snap build process, the snap downloads the CUDA keyring deb package using `curl` over HTTPS and verifies its integrity using SHA256 checksums.
```
Comment (aieri) [inline snap/snapcraft.yaml:69]:
```suggestion
  During the snap build process, snapcraft downloads the CUDA keyring deb package using `curl` over HTTPS and verifies its integrity using SHA256 checksums.
```

**Review by aieri — CHANGES_REQUESTED** [review]
last one!

**Review by gabrielcocenza — APPROVED** [review]

**Pull request merged**